File: fixtures/scale/python/large/cipherlab/checks/check_mixcrc.py
from cipherlab.ciphers.mixcrc import mixcrc_digest
from cipherlab.specs.mixcrc_spec import mixcrc_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_mixcrc():
    spec = mixcrc_spec()
    for text in corpus_hash():
        first = mixcrc_digest(text)
        second = mixcrc_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"

File: fixtures/scale/ruby/large/lib/check_mixcrc.rb
require_relative 'mixcrc_digest'
require_relative 'mixcrc_digest_pair'
require_relative 'corpus_hash'

def check_mixcrc
  corpus_hash.each do |text|
    first = mixcrc_digest(text)
    second = mixcrc_digest(text)
    return false if first != second || first.length != 8
  end
  true
end
